# Andmetega tutvumine

Selles notebook'is analüüsime toorandmestikku.

## 1. Andmete laadimine ja põhiinfo

In [20]:
import pandas as pd

# Laadime andmed
df = pd.read_csv('andmed/toorandmed.csv')

print(f"Andmepunktide arv (ridade arv): {df.shape[0]}")
print(f"Tunnuste arv (veergude arv): {df.shape[1]}")
print(f"\nAndmestiku dimensioonid: {df.shape}")

Andmepunktide arv (ridade arv): 3031
Tunnuste arv (veergude arv): 223

Andmestiku dimensioonid: (3031, 223)


## 2. Ühe andmerea põhjalik vaatlus

Vaatame 256. ainekursust detailselt läbi.

In [21]:
# Valime 456. rea (456. aine)
valitud_aine = df.iloc[256]

print("=" * 80)
print("256. AINE DETAILNE INFO")
print("=" * 80)
print()

# Kuvame kõik andmed selle aine kohta
for tunnus, vaartus in valitud_aine.items():
    print(f"{tunnus}:")
    print(f"  {vaartus}")
    print()

256. AINE DETAILNE INFO

course_uuid:
  6507f899-5280-9f02-e984-079bb91bcef2

code:
  LOKT.00.023

title__en:
  Practical Speciality Training

version__title__en:
  Practical Speciality Training

credits:
  6.0

overview__description__en:
  Aim of the course is to give students first practical experience of working as a scientist or laboratory assistant in the factory or scientific laboratory. During the practical part both theoretical and practical experience on the specific field will be obtained as at least one-month full job in particular enterprise. Obtained theoretical and practical experience will be summarized in written report. The project has to be defended in open discussion of seminar classes.

version__overview__description__en:
  Aim of the course is to give students first practical experience of working as a scientist or laboratory assistant in the factory or scientific laboratory. During the practical part both theoretical and practical experience on the specific field 

In [22]:
# Kokkuvõte 256. ainest põhitunnuste kohta
print("=" * 80)
print("256. AINE PÕHIINFO KOKKUVÕTE")
print("=" * 80)
print()
print(f"Ainekood: {valitud_aine['code']}")
print(f"Aine nimetus (EN): {valitud_aine['title__en']}")
print(f"Aine nimetus (ET): {valitud_aine['title__et']}")
print(f"Ainepunktid (EAP): {valitud_aine['credits']}")
print(f"Olek: {valitud_aine['state__et']} ({valitud_aine['state__en']})")
print(f"Viimane uuendus: {valitud_aine['last_update']}")


256. AINE PÕHIINFO KOKKUVÕTE

Ainekood: LOKT.00.023
Aine nimetus (EN): Practical Speciality Training
Aine nimetus (ET): Praktika
Ainepunktid (EAP): 6.0
Olek: Kinnitatud (confirmed)
Viimane uuendus: 2024-02-05T11:04:36


## 3. Kõik tunnused (veerud) andmestikus

In [23]:
# Näitame kõiki veergude nimesid
print(f"Kokku {len(df.columns)} tunnust:\n")

for i, veerg in enumerate(df.columns, 1):
    print(f"{i:3d}. {veerg}")

Kokku 223 tunnust:

  1. course_uuid
  2. code
  3. title__en
  4. version__title__en
  5. credits
  6. overview__description__en
  7. version__overview__description__en
  8. overview__learning_outcomes_text_en
  9. uuid
 10. state__code
 11. state__et
 12. state__en
 13. last_update
 14. title__et
 15. general__input_languages
 16. general__structural_unit_shares
 17. general__type__code
 18. general__type__et
 19. general__type__en
 20. general__structural_units
 21. additional_info__duration_in_semesters
 22. additional_info__assessment_scale__code
 23. additional_info__assessment_scale__et
 24. additional_info__assessment_scale__en
 25. additional_info__is_vota_course
 26. additional_info__is_continuous_learning_course
 27. additional_info__prerequisites
 28. overview__description__et
 29. overview__objectives
 30. overview__learning_outcomes
 31. _actions__is_editable
 32. _actions__is_version_creatable
 33. _actions__is_deletable
 34. overview__objectives_text_en
 35. version__uu

Nimi: Jakko Turro  
Valitud aine nimi: Praktika  
Valitud aine kood: LOKT.00.023  
Lühike ülevaade vormistusest ja teistest tähelepanekutest:  
Andmerida sisaldab praktikaaine andmeid meditsiini valdkonnas. 92 tunnust 223-st on tühjad (NaN väärtused), mis näitab andmete mittetäielikkust. 14 tunnust sisaldavad JSON-struktureeritud andmeid (struktuuriüksused, õppejõud, eeldusained). EAP on float64 tüüpi (6.0), kuupäevad järgivad ISO 8601 formaati. UUID-d on korrektselt vormindatud. Andmetes leidub korduvat informatsiooni erinevates keeltes ja versioonides, mis suurendab andmemahtu. Kahtlane on väga üldine eestikeelne aine nimetus "Praktika" võrreldes detailsema ingliskeelse nimega "Practical Speciality Training", mis võib viidata ebatäpsele tõlkele või andmete sisestamise vigadele. Samuti on 41.3% andmetest puudu, mis on ebatavaliselt suur osakaal ja võib mõjutada andmete kasutamist analüüsides.
